backends/ebpf/tests/ptf/common.py

```diff
@@ -361,24 +357,32 @@ def table_verify(self, table, key, action=0, priority=None, data=None, reference
     def action_selector_add_action(self, selector, action, data=None):
         cmd = "psabpf-ctl action-selector add-member pipe {} {} ".format(TEST_PIPELINE_ID, selector)
         cmd = cmd + self._table_create_str_from_action(action)
-        if data:
-            cmd = cmd + " data"
-            for d in data:
-                cmd = cmd + " {}".format(d)
+        cmd = cmd + self._table_create_str_from_data(data=data, counters=None, meters=None)
         _, stdout, _ = self.exec_ns_cmd(cmd, "ActionSelector add-member failed")
-        return int(stdout)
+        return json.loads(stdout)[selector]["added_member_ref"]
 
     def action_selector_create_empty_group(self, selector):
         cmd = "psabpf-ctl action-selector create-group pipe {} {}".format(TEST_PIPELINE_ID, selector)
         _, stdout, _ = self.exec_ns_cmd(cmd, "ActionSelector create-group failed")
-        return int(stdout)
+        return json.loads(stdout)[selector]["added_group_ref"]
 
     def action_selector_add_member_to_group(self, selector, group_ref, member_ref):
         cmd = "psabpf-ctl action-selector add-to-group pipe {} {} {} to {}"\
             .format(TEST_PIPELINE_ID, selector, member_ref, group_ref)
         self.exec_ns_cmd(cmd, "ActionSelector add-to-group failed")
 
+    def action_profile_add_action(self, ap, action, data=None):
+        cmd = "psabpf-ctl action-profile add-member pipe {} {} ".format(TEST_PIPELINE_ID, ap)
+        cmd = cmd + self._table_create_str_from_action(action)
+        cmd = cmd + self._table_create_str_from_data(data=data, counters=None, meters=None)
+        _, stdout, _ = self.exec_ns_cmd(cmd, "ActionProfile add-member failed")
+        return json.loads(stdout)[ap]["added_member_ref"]
+
     def digest_get(self, name):
+        # Possible race conditions: many test cases may send packet and just
+        # after that try to read digest message. In rare case packet can be
+        # not yet processed so digest is not yet available.
+        time.sleep(0.1)
```

Comment:
Is it enough? I think it's OK to set higher value in tests to guarantee successful execution, if needed.

Reply:
I have seen once such race condition, but I were never able to reproduce it. These tests were passed many times without problems, so I think 100 ms is enough.